socket_linux.cc:  Switch back to using accept instead of accept4

--- a/dart/runtime/bin/socket_linux.cc	2014-04-28 08:47:39.189138056 -0700
+++ b/dart/runtime/bin/socket_linux.cc	2014-04-28 09:43:59.954574690 -0700
@@ -420,8 +420,7 @@ intptr_t ServerSocket::Accept(intptr_t f
   intptr_t socket;
   struct sockaddr clientaddr;
   socklen_t addrlen = sizeof(clientaddr);
-  socket = TEMP_FAILURE_RETRY(accept4(
-      fd, &clientaddr, &addrlen, SOCK_NONBLOCK | SOCK_CLOEXEC));
+  socket = TEMP_FAILURE_RETRY(accept(fd, &clientaddr, &addrlen));
   if (socket == -1) {
     if (IsTemporaryAcceptError(errno)) {
       // We need to signal to the caller that this is actually not an
@@ -430,6 +429,9 @@ intptr_t ServerSocket::Accept(intptr_t f
       ASSERT(kTemporaryFailure != -1);
       socket = kTemporaryFailure;
     }
+  } else {
+    Socket::SetNonBlocking(socket);
+    FDUtils::SetCloseOnExec(socket);
   }
   return socket;
 }
@@ -446,6 +448,11 @@ void Socket::Close(intptr_t fd) {
 }
 
 
+bool Socket::SetNonBlocking(intptr_t fd) {
+  return FDUtils::SetNonBlocking(fd);
+}
+
+
 bool Socket::GetNoDelay(intptr_t fd, bool* enabled) {
   int on;
   socklen_t len = sizeof(on);
